### ISSUE TITLE
Looping when we use the wildcard regex like (.*)

### ISSUE THREAD

**Issue by rahulnatarajan98** [lonekorean/highlight-within-textarea#40]
The Ranges for regular expression makes an infinite loop while trying to use wildcard regex (.*)

**Comment by ElonSchliessMuskel** [comment]
Same issue here and still not fixed. Unfortunately the script is not usable in this state, because the whole tab crashes, even if the highlight function is called within a try/catch block. 

@rahulnatarajan98 have you already found a solution or do you know exactly what is causing the error? Then maybe we could check if the regex fulfills this crash condition and then not call the highlight function afterwards.

**Comment by rahulnatarajan98** [comment]
Yeah. I created a if statement stating the regex exactly matches with ".*" then ranges array should have the index starting from 0 to length of the matching string.

Also, we should run the while loop whenever the regex value is not equal to .*

Doing this is not a quite good idea. But this will overcome the issue.

**Comment by ElonSchliessMuskel** [comment]
Okay, I see. Thank you. The problem with this approach is that there are an infinite number of regular expressions that will cause a crash. 
A random example: `/(?!thiswillcrash)/g`
So a blacklist of "bad" patterns is not very useful. For my project I need a general solution. Until the problem is not solved, I have to check somehow in advance if the regex can cause a crash and not call the highlight function afterwards. Such a general way exists: If you insert the expression into regexr.com, a warning will appear saying that this script can cause an infinite number of matches. 

Since the tab crashes all the time and the devtools disconnect immediately, it is difficult to investigate the error. So I can only assume that this is the problem. I was hoping that you have already taken a closer look at the problem or found a general solution.

**Comment by rahulnatarajan98** [comment]
Actually, I had validation checks before using the regex in the plugin. I used the RegExp Object in Javascript in creating a regex from a string. For example
`var regex = new RegExp('(?thiswillcrash)')`

In JS, this regex wont get created as it in invalid regular expression.
I also made the plugin to run for valid regular expressions.
